automerge-backend/src/backend.rs

```diff
@@ -18,6 +18,7 @@ pub struct Backend {
     actors: ActorMap,
     hashes: HashMap<amp::ChangeHash, Change>,
     history: Vec<amp::ChangeHash>,
+    history_idx: HashMap<amp::ChangeHash, usize>,
```

Comment:
I feel like there is some redundancy here, could we instead move to something like:
```rust
history: Vec<Change>,
history_index: HashMap<amp::ChangeHash, usize>,
```
and drop the `hashes` field?

And nitpicking, I prefer using whole words, e.g. index instead of idx but that is very minor.

Reply:
I agree!  hashes has been removed